sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -116,24 +120,14 @@ export class InteractiveBrowserCredential implements TokenCredential {
     // eslint-disable-next-line
     return new Promise<AccessToken | null>(async (resolve, reject) => {
       // eslint-disable-next-line
-      let listen: http.Server | undefined;
       let socketToDestroy: Socket | undefined;
 
-      function cleanup(): void {
-        if (listen) {
-          listen.close();
-        }
-        if (socketToDestroy) {
-          socketToDestroy.destroy();
-        }
-      }
-
       // Create Express App and Routes
-      const app = express();
 
-      app.get("/", async (req, res) => {
+      const requestListener = async (req: http.IncomingMessage, res: http.ServerResponse) => {
+        const url = new URL(req.url!, this.redirectUri);
```

Comment:
do we ever have to be worried about `req.url` being undefined? Are the typings just bad? What would happen in the failure case, `URL` constructor throwing? 

Actually what even happens if requestListener fails? Since it's an async function it will reject, does Node do something with that rejection? Since all that seems to happen is the [function gets registered on the `request` event](https://nodejs.org/dist/latest-v14.x/docs/api/http.html#http_http_createserver_options_requestlistener), I don't think it would, so Node would hit an "unhandled rejection" error and possibly die.

Reply:
Looks like handling for async functions as event handlers (and actually subscribing to their rejection) is still very experimental: https://nodejs.org/api/events.html#events_capture_rejections_of_promises

Reply:
@xirzec - can you double check the new approach before I land it?

Reply:
Taking a look now!